sky/shell/platform/android/org/domokit/sky/shell/PlatformViewAndroid.java

```diff
@@ -291,4 +316,315 @@ void runFromBundle(String path) {
     private static native void nativeSurfaceCreated(long nativePlatformViewAndroid,
                                                     Surface surface);
     private static native void nativeSurfaceDestroyed(long nativePlatformViewAndroid);
+
+
+    // ACCESSIBILITY
+
+    @Override
+    protected void onAttachedToWindow() {
+        super.onAttachedToWindow();
+        if (mAccessibilityManager.isEnabled() || mAccessibilityManager.isTouchExplorationEnabled())
+          ensureAccessibilityEnabled();
+        mAccessibilityManager.addAccessibilityStateChangeListener(this);
+        mAccessibilityManager.addTouchExplorationStateChangeListener(this);
+    }
+
+    @Override
+    public void onAccessibilityStateChanged(boolean enabled) {
+        if (enabled)
+            ensureAccessibilityEnabled();
+    }
+
+    @Override
+    public void onTouchExplorationStateChanged(boolean enabled) {
+        if (enabled)
+            ensureAccessibilityEnabled();
+        // TODO(ianh): else, actually discard the state for exploration
+    }
+
+    private FlutterSemanticsToAndroidAccessibilityBridge mAccessibilityNodeProvider;
+    private SemanticsServer.Proxy semanticsServer;
+
+    void ensureAccessibilityEnabled() {
+        if (mAccessibilityNodeProvider == null) {
+            mAccessibilityNodeProvider = new FlutterSemanticsToAndroidAccessibilityBridge();
+            Core core = CoreImpl.getInstance();
+            Pair<SemanticsServer.Proxy, InterfaceRequest<SemanticsServer>> server =
+                      SemanticsServer.MANAGER.getInterfaceRequest(core);
+            semanticsServer = server.first;
+            mDartServiceProvider.connectToService("semantics::SemanticsServer", server.second.passHandle());
+            semanticsServer.addSemanticsListener(mAccessibilityNodeProvider);
+        }
+        assert semanticsServer != null;
+    }
+
+    @Override
+    public AccessibilityNodeProvider getAccessibilityNodeProvider() {
+        ensureAccessibilityEnabled();
+        return mAccessibilityNodeProvider;
+    }
+
+    // TODO(ianh): implement touch exploration
+
+    // TODO(ianh): implement accessibility focus
+
+    void resetAccessibilityTree() {
+        if (mAccessibilityNodeProvider != null)
+            mAccessibilityNodeProvider.reset();
+    }
+
+    public class FlutterSemanticsToAndroidAccessibilityBridge extends AccessibilityNodeProvider
```

Comment:
Sure. I'd have to add a pointer to the View, but that's fine. (Right now I'm using the fact that they're Java-esque inner classes and thus have pointers to the parent classes.)